src/main/kotlin/sh/hsp/techtree/FileSystemYamlReader.kt

```diff
@@ -0,0 +1,19 @@
+package sh.hsp.techtree
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
+import com.fasterxml.jackson.module.kotlin.readValue
+import com.fasterxml.jackson.module.kotlin.registerKotlinModule
+import java.io.File
+import java.net.URL
+
+class FileSystemYamlReader : YamlReader {
+    override fun readToModel(resource: URL): TreeModel {
+        val mapper = ObjectMapper(YAMLFactory()).registerKotlinModule()
```

Comment:
Spoko, tylko może bym tego object mappera podał w konstruktorze

Reply:
Jak wprowadzisz tego mappera do konstruktora to później chce returna. Stąd nie wiedziałem jak to zmienić

Reply:
```
class FileSystemYamlReader(private val mapper: ObjectMapper) : YamlReader {
    override fun readToModel(resource: URL): TreeModel {
        val file = File(resource.toURI())
        val yamlData = file.readText()

        return mapper.readValue(yamlData)
    }

}
```
tak powinno działać